### PR TITLE
added script to run influence analysis

### LINKED ISSUE #129: Refactor code for influence analysis so it works outside OpenAlex

### PR DESCRIPTION
**Pull request by Juan-Mateos**
Closes #129  

I have refactored the code used for the OpenAlex influence analysis into a dataset-agnostic set of functions that you can find in `ai_genomics/analysis/influence/influence_analysis.py`. You can run that script to test the functions on the gtr data.

---

Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [x] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [x] I have explained the feature in this PR or (better) in `output/reports/`
- [x] I have requested a code review

### CODE REVIEW

File: ai_genomics/analysis/influence/influence_analysis.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        tm.create_topic_names(lda_model, num_topics),
+        list(text_dict.keys()),
+    )
+
+
+def calculate_topic_shares(
+    topic_df: pd.DataFrame,
+    disc_names_ids: Dict[str, List],
+    baseline_discs: List[str] = ["ai", "genomics"],
+    binarise_thres: float = 0.9,
```
Comment (Jack-Vines) [inline ai_genomics/analysis/influence/influence_analysis.py:82]: Any particular reason for 0.9 being chosen as threshold?

Comment (Jack-Vines) [reply ai_genomics/analysis/influence/influence_analysis.py:82]: Maybe worth adding to the docstring

File: ai_genomics/analysis/influence/influence_analysis.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        .sum()
+    )
+
+
+if __name__ == "__main__":
+    logging.info("testing with the GtR data")
+
+    import ai_genomics.getters.gtr as gtr
+
+    gtr_meta = gtr.get_ai_genomics_project_table()
```
Comment (Jack-Vines) [inline ai_genomics/analysis/influence/influence_analysis.py]: just to note, this initially failed for me as the default in the getter is local. Not really for this PR but i suppose there's an assumption that whoever is runnning this has run all steps up to this point and so has all output files locally, that we should note/clean up further down the line

File: ai_genomics/analysis/influence/influence_analysis.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+    # This calculates the influence of a discipline in a paper based
+    # on its topic distribution
+    disc_score = calculate_influence_score(gtr_topic_df, topic_shares)
+
+    gtr_meta_sampled = gtr_meta_sampled.merge(
+        disc_score.reset_index(drop=False), left_on="id", right_on="doc_id"
+    )
+
+    logging.info("abstracts for projects influenced by AI")
```
Comment (Jack-Vines) [inline ai_genomics/analysis/influence/influence_analysis.py:209]: This logging is all really useful

File: ai_genomics/analysis/influence/influence_analysis.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+        ]
+    )
+
+    gtr_text = {
+        _id: text
+        for _id, text in zip(gtr_meta_sampled["id"], gtr_meta_sampled["abstract_text"])
+    }
+
+    gtr_proc = text_processing(gtr_text)
+    gtr_topic_df = make_topic_distribution(gtr_proc, num_topics=200)
```
Comment (Jack-Vines) [inline ai_genomics/analysis/influence/influence_analysis.py:158]: Why 200 topics here vs the default of 100 in the function?

**Review by Jack-Vines — APPROVED** [review]
It all works great! I've left a few points of clarification but mostly just curiosity around parameters

**Comment by Juan-Mateos** [comment]
Thanks @Jack-Vines ! 

I have added `bin_thres` to the docstrings and made `local=False` in the gtr project getter.

Regarding your questions.
* I chose 0.9 heuristically. It would be nice to tune it but I am not sure about the validation metric. In any case, if the topic weights in a topic are similarly distributed across disciplines (AI or genomics), changing the threshold should't impact the results (lower thresholds will lead to the inclusion of more papers as "having a topic" but the relative shares of projects over disciplines shouldn't change. Perhaps we could test this.
* 200 topics is a hangover from when I was trying to run the analysis with unbalanced discipline mixes (~2x AI as genomics) - I increased resolution to increase the presence of genomics papers vs what you get with 100 topics. I ended dealing with the imbalance through sampling so I could change this back to 100.

**Review by Jack-Vines — APPROVED** [review]

**Pull request merged**